devicetypes/fibargroup/fibaro-flood-sensor-zw5.src/fibaro-flood-sensor-zw5.groovy

```diff
@@ -1,301 +1,539 @@
+			sendEvent(name: "syncStatus", value: "incomplete")
+		}
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.manufacturerspecificv2.ManufacturerSpecificReport cmd) {
+	log.debug "ManufacturerSpecificReport"
+	log.debug "manufacturerId:	 ${cmd.manufacturerId}"
+	log.debug "manufacturerName: ${cmd.manufacturerName}"
+	log.debug "productId:		 ${cmd.productId}"
+	log.debug "productTypeId:	 ${cmd.productTypeId}"
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.manufacturerspecificv2.DeviceSpecificReport cmd) {
+	log.debug "DeviceSpecificReport"
+	log.debug "deviceIdData:				${cmd.deviceIdData}"
+	log.debug "deviceIdDataFormat:			${cmd.deviceIdDataFormat}"
+	log.debug "deviceIdDataLengthIndicator: ${cmd.deviceIdDataLengthIndicator}"
+	log.debug "deviceIdType:				${cmd.deviceIdType}"
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.versionv1.VersionReport cmd) {
+	log.debug "VersionReport"
+	log.debug "applicationVersion:		${cmd.applicationVersion}"
+	log.debug "applicationSubVersion:	${cmd.applicationSubVersion}"
+	log.debug "zWaveLibraryType:		${cmd.zWaveLibraryType}"
+	log.debug "zWaveProtocolVersion:	${cmd.zWaveProtocolVersion}"
+	log.debug "zWaveProtocolSubVersion: ${cmd.zWaveProtocolSubVersion}"
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.batteryv1.BatteryReport cmd) {
+	log.debug "BatteryReport"
+	log.debug "cmd: "+cmd
+	log.debug "location: "+location
+	def timeDate = location.timeZone ? new Date().format("yyyy MMM dd EEE h:mm:ss a", location.timeZone) : new Date().format("yyyy MMM dd EEE h:mm:ss")
+	def value = cmd.batteryLevel
+	def desc = "Curent battery level"
+	def isStateChange = false
+
+	if (value == 0xFF) {  // Special value for low battery alert
+		value = 1
+		desc = "${device.displayName} has a low battery"
+		isStateChange = true
+	}
+	sendEvent(name: "battery", value: value, descriptionText: desc, isStateChange: isStateChange)
+	sendEvent(name: "batteryStatus", value: "Battery: $cmd.batteryLevel%\n($timeDate)")
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.notificationv3.NotificationReport cmd) {
+	log.debug "NotificationReport"
+	def map = [:]
+	def alarmInfo = "Last alarm detection: "
+	if (cmd.notificationType == 5) {
+		switch (cmd.event) {
+			case 2:
+				map.name = "water"
+				map.value = "wet"
+				map.descriptionText = "${device.displayName} is ${map.value}"
+				state.lastAlarmDate = "\n"+new Date().format("yyyy MMM dd EEE HH:mm:ss")
+				//state.lastAlarmDate = "\n"+new Date().format("yyyy MMM dd EEE HH:mm:ss", location.timeZone)
+				multiStatusEvent(alarmInfo + state.lastAlarmDate)
+				break
+
+			case 0:
+				map.name = "water"
+				map.value = "dry"
+				map.descriptionText = "${device.displayName} is ${map.value}"
+				multiStatusEvent(alarmInfo + state.lastAlarmDate)
+				break
+		}
+	} else if (cmd.notificationType == 7) {
+		switch (cmd.event) {
+			case 0:
+				map.descriptionText = "${device.displayName}: tamper alarm has been deactivated"

```

Comment:
Just setting the description text is probably not what was intended here. I think the tamper events should be put in the map, while the batteryStatus event should be sent out of band.